pkg/cloud/azure/assets/cloud-controller-manager-deployment.yaml

```diff
@@ -4,17 +4,20 @@ metadata:
   name: azure-cloud-controller-manager
   namespace: openshift-cloud-controller-manager
   labels:
-    app: azure-cloud-controller-manager
+    k8s-app: azure-cloud-controller-manager
+    infrastructure.openshift.io/cloud-controller-manager: {{ .cloudproviderName }}
 spec:
   selector:
     matchLabels:
-      app: azure-cloud-controller-manager
+      k8s-app: azure-cloud-controller-manager
```

Comment:
This change will break cluster upgrades, labels on deploymenys are immutable once created. Is there a way we can test this, will we need to delete and recreate to achieve this?

Reply:
https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/174/commits/0e963afcdeb0d3ead499d4abc7d67911801bc943 - recreate logic with envtest based tests